hbase-common/src/main/java/org/apache/hadoop/hbase/HBaseConfiguration.java

```diff
@@ -77,6 +81,39 @@ private static void checkDefaultsVersion(Configuration conf) {
     }
   }
 
+  /**
+   * The hbase.ipc.server.reservoir.initial.max and hbase.ipc.server.reservoir.initial.buffer.size
+   * were introduced in HBase2.0.0, while in HBase3.0.0 the two config keys will be replaced by
+   * hbase.server.allocator.max.buffer.count and hbase.server.allocator.buffer.size.
+   * Also the hbase.ipc.server.reservoir.enabled will be replaced by
+   * hbase.server.allocator.pool.enabled. Keep the three old config keys here for HBase2.x
+   * compatibility.
+   * <br>
+   * HBASE-24667: This config hbase.regionserver.hostname.disable.master.reversedns will be
+   * replaced by hbase.unsafe.regionserver.hostname.disable.master.reversedns. Keep the old config
+   * keys here for backward compatibility.
+   */
+  private static void addDeprecatedKeys() {
+    Configuration.addDeprecations(new DeprecationDelta[]{
+      new DeprecationDelta("hbase.regionserver.hostname", "hbase.unsafe.regionserver.hostname"),
+      new DeprecationDelta("hbase.regionserver.hostname.disable.master.reversedns",
+        "hbase.unsafe.regionserver.hostname.disable.master.reversedns"),
+      new DeprecationDelta("hbase.offheapcache.minblocksize",
+        "hbase.blockcache.minblocksize"),
+      new DeprecationDelta("hbase.ipc.server.reservoir.enabled",
+        "hbase.server.allocator.pool.enabled"),
+      new DeprecationDelta("hbase.ipc.server.reservoir.initial.max",
+        "hbase.server.allocator.max.buffer.count"),
+      new DeprecationDelta("hbase.ipc.server.reservoir.initial.buffer.size",
+        "hbase.server.allocator.buffer.size"),
+      new DeprecationDelta("hlog.bulk.output", "wal.bulk.output"),
+      new DeprecationDelta("hlog.input.tables", "wal.input.tables"),
+      new DeprecationDelta("hlog.input.tablesmap", "wal.input.tablesmap"),
+      new DeprecationDelta("hbase.master.mob.ttl.cleaner.period",
+        "hbase.master.mob.cleaner.period")
```

Comment:
branch-2 doesn't have this configuration, so it needs to be deleted. The rest doesn't need to change.

Reply:
Like this?

Reply:
Yes, there is nothing else to modify.